packages/ti_maltiverse/data_stream/indicator/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,236 @@
+      value: "https://maltiverse.com/ip/{{ threat.indicator.ip }}"
+      if: 'ctx.json?.type == "ip"'
+## TODO: match the AS regex
+  - rename:
+      field: json.city
+      target_field: threat.indicator.geo.city_name
+      ignore_missing: true
+      if: 'ctx.json?.type == "ip" && ctx.json?.city != null'
+  - rename:
+      field: json.country_code
+      target_field: threat.indicator.geo.country_iso_code
+      ignore_missing: true
+      if: 'ctx.json?.type == "ip" && ctx.json?.country_code != null'
+  - rename:
+      field: json.country_code
+      target_field: threat.indicator.geo.country_iso_code
+      ignore_missing: true
+      if: 'ctx.json?.type == "ip" && ctx.json?.country_code != null'
+  # IP geo fields
+  - convert:
+      field: json.location.lat
+      target_field: threat.indicator.geo.location.lat
+      type: double
+      if: "ctx.json?.location?.lat != null && ctx.json?.location?.lon != null"
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Cannot convert lat field "{{{ json.lat }}}" to double'
+  - convert:
+      field: json.location.lon
+      target_field: threat.indicator.geo.location.lon
+      type: double
+      if: "ctx.json?.location?.lat != null && ctx.json?.location?.lon != null"
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Cannot convert lon field "{{{ json.lon }}}" to double'
+
+## hostname fields
+  - rename:
+      field: json.hostname
+      target_field: threat.indicator.url.domain
+      ignore_missing: true
+      if: 'ctx.json?.type == "hostname"'
+  - rename:
+      field: json.domain
+      target_field: threat.indicator.url.registered_domain
+      ignore_missing: true
+      if: 'ctx.json?.type == "hostname"'
+  - rename:
+      field: json.tld
+      target_field: threat.indicator.url.top_level_domain
+      ignore_missing: true
+      if: 'ctx.json?.type == "hostname"'
+  - set:
+      field: threat.indicator.reference
+      value: "https://maltiverse.com/hostname/{{ threat.indicator.url.domain }}"
+      if: 'ctx.json?.type == "hostname"'
+## TODO: match the AS regex
+## URL fields
+  - rename:
+      field: json.url
+      target_field: threat.indicator.url.full
+      ignore_missing: true
+      if: 'ctx.json?.type == "url"'
+  - rename:
+      field: json.url
+      target_field: threat.indicator.url.original
+      ignore_missing: true
+      if: 'ctx.json?.type == "url"'
+  - rename:
+      field: json.tld
+      target_field: threat.indicator.url.top_level_domain
+      ignore_missing: true
+      if: 'ctx.json?.type == "url" && ctx.json?.tld != null'
+  - rename:
+      field: json.domain
+      target_field: threat.indicator.url.registered_domain
+      ignore_missing: true
+      if: 'ctx.json?.type == "url" && ctx.json?.domain != null'
+  - set:
+      field: threat.indicator.reference
+      value: "https://maltiverse.com/url/{{ json.urlchecksum }}"
+      if: 'ctx.json?.type == "url"'
```

ctx.json?.type == "url"
```suggestion
      if: 'ctx.json?.type == "url" && ctx.json?.urlchecksum != null'  
```